otsdaq/FECore/FEVInterface.cc

```diff
@@ -351,19 +362,37 @@ try
 
 
 			// Use artdaq Metric Manager if available,
-			if(channel->monitoringEnabled_ && metricMan && metricMan->Running() && universalAddressSize_ <= 8)
+			if(channel->monitoringEnabled_ && metricMan && metricMan->Running() && universalAddressSize_ <= 8) 
 			{
 				uint64_t val = 0;  // 64 bits!
 				for(size_t ii = 0; ii < universalAddressSize_; ++ii)
 					val += (uint8_t)readVal[ii] << (ii * 8);
 
-				__FE_COUT__ << "Sending sample to Metric Manager..." << __E__;
-				metricMan->sendMetric(channel->fullChannelName_, val, "", 3, artdaq::MetricMode::LastPoint);
-			}
-			else
+				// Unit transforms
+				if((channel->transformation_).size() > 1) // Execute transformation if a formula is present
+				{ 
+					__FE_COUT__ << "Transformation formula = " <<channel->transformation_ << __E__;
+                
+					TFormula transformationFormula("transformationFormula", (channel->transformation_).c_str());
+					double transformedVal = transformationFormula.Eval(val);
```

Comment:
Probably a good idea to check for NaN here, instead of just reporting the value

Reply:
I added an `isnan` check for `transformedVal`, which throws a `COUT_ERR` if it fails.